### PR TITLE
Feature/enable piwik tracking

### PR DESCRIPTION
**Pull request by achen2401**
part of https://www.pivotaltracker.com/n/projects/2493717/stories/183594028
enable PIWIK tracking if matomo site ID is configured.

### CODE REVIEW

File: patientsearch/src/js/helpers/utility.js [hunk 1]
```diff
@@ -393,3 +404,28 @@ export function putPatientData(patientId, data, errorCallback, successCallback)
     if (successCallback) successCallback();
   });
 }
+
+export function addMamotoTracking(siteId, userId) {
+  if (document.querySelector("#matomoScript")) return;
```
Comment (pbugni) [inline patientsearch/src/js/helpers/utility.js:409]: It isn't obvious to me where you're checking for a siteId of null (None) as would be the case if `MATOMO_SITE_ID` isn't defined?

Comment (achen2401) [reply patientsearch/src/js/helpers/utility.js:409]: thanks @pbugni  for catching this, fixed.

File: patientsearch/src/js/components/PatientListTable.js [hunk 2]
```diff
@@ -1093,6 +1095,11 @@ export default function PatientListTable() {
           return false;
         }
         if (appSettings) {
+          addMamotoTracking(
+            appSettings["MATOMO_SITE_ID"],
+            // use preferred_username from token as user id
```
Comment (ivan-c) [inline patientsearch/src/js/components/PatientListTable.js:1100]: Just a small change to remove some ambiguity between OIDC/SoF and Access/ID token (These attributes are provided by flask-oidc from decoded JWTs)
```suggestion
            // use preferred_username from OIDC ID token as user id
```

**Review by ivan-c — APPROVED** [review]
Looks good!

**Review by mcjustin — APPROVED** [review]

**Comment by mcjustin** [comment]
> @mcjustin do you want to have a look at this PR before I merge this?

Looks great, thanks @achen2401

**Pull request merged**